lib/Sema/TypeCheckDecl.cpp

```diff
@@ -242,31 +240,31 @@ void TypeChecker::validateWhereClauses(ProtocolDecl *protocol,
 /// to which this type declaration conforms.
 void TypeChecker::checkInheritanceClause(Decl *decl,
                                          GenericTypeResolver *resolver) {
-  TypeResolutionOptions options;
+  TypeResolutionOptions options = None;
   DeclContext *DC;
   if (auto nominal = dyn_cast<NominalTypeDecl>(decl)) {
     DC = nominal;
-    options |= TypeResolutionFlags::GenericSignature;
-    options |= TypeResolutionFlags::InheritanceClause;
+    options = TypeResolutionOptions(TypeResolverContext::GenericSignature);
+    options.setContext(TypeResolverContext::InheritanceClause);
```

Comment:
What does it mean to set the context twice like this?

Reply:
Unless I'm missing something, `TypeResolutionOptions` is about providing context to nested type resolvers. Some type resolvers need to know the current context, some need to know the outermost "base" context, and some need to know both. After this pull request, the `TypeResolutionOptions` constructor sets both the base and current contexts to the provided `TypeResolverContext`. When `setContext()` is called, only the current context is updated. 

In the case of this specific code, `GenericSignature` doesn't make sense as the sub-context of another `TypeResolverContext` and therefore it is setup as the "base" context in this example. That being said, we need to convey that the inheritance clause needs to be checked too, and therefore it is set as the "current" context so that it can be special cased accordingly.

Reply:
I see, thanks. It looks like InheritanceClause is only checked in one place — and there’s a FIXME about it. Might be worth trying to remove it altogether.

Reply:
Weird. I commented out the exception and nothing broke on my Linux box. Is that expected?